code/src/components/Header.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+import { format } from 'date-fns'
+
+import { tasks } from '../reducers/tasks'
+import { HeaderContainer, HeaderDate, HeaderTitleDateContainer, HeaderCounterButtonContainer, MainHeader, CompletedTasksText, AllDoneButton } from './styled components/HeaderStyling'
+
+const Header = () => {
+    const dispatch = useDispatch()
+
+    const totalNumberOfTasks = useSelector((store) => (
+        store.tasks.items.length
+    ))
+
+     const completedTasks = useSelector((store) => (
+        store.tasks.items.filter((task => task.completed)).length
+    ))
+
+    const onClickClearAll = () => {
+        dispatch(tasks.actions.clearAllTasks())
+    }
+
+    return (
+        <HeaderContainer>
+            <HeaderTitleDateContainer>
+                <MainHeader>Todo</MainHeader>
+                <HeaderDate>{format(new Date(), 'MMMM do')}</HeaderDate>
+            </HeaderTitleDateContainer>
+            <HeaderCounterButtonContainer>
+                <CompletedTasksText>
+                    Completed: {completedTasks}/{totalNumberOfTasks}
+                </CompletedTasksText>
+                <AllDoneButton onClick={onClickClearAll}>
+                    All done!
+                </AllDoneButton>
+            </HeaderCounterButtonContainer>
+        </HeaderContainer>
+    )
+}
```

Comment:
I like how you have structured your styled components in separate files and then imported them. Next time i will copy you and try to do the same!